files/default/sql/ddl/updates/2.1.0.sql

```diff
@@ -147,4 +147,65 @@ CREATE TABLE `on_demand_option` (
-) ENGINE=ndbcluster DEFAULT CHARSET=latin1 COLLATE=latin1_general_cs;
\ No newline at end of file
+) ENGINE=ndbcluster DEFAULT CHARSET=latin1 COLLATE=latin1_general_cs;
+
+CREATE TABLE `statistics_config` (
+  `id` int(11) NOT NULL AUTO_INCREMENT,
+  `feature_group_id` int(11),
+  `training_dataset_id` int(11),
+  `descriptive` TINYINT(1) NOT NULL DEFAULT '1',
+  `correlations` TINYINT(1) NOT NULL DEFAULT '0',
+  `histograms` TINYINT(1) NOT NULL DEFAULT '0',
+  PRIMARY KEY (`id`),
+  KEY `feature_group_id` (`feature_group_id`),
+  KEY `training_dataset_id` (`training_dataset_id`),
+  CONSTRAINT `fg_statistics_config_fk` FOREIGN KEY (`feature_group_id`) REFERENCES `feature_group` (`id`) ON DELETE CASCADE ON UPDATE NO ACTION,
+  CONSTRAINT `td_statistics_config_fk` FOREIGN KEY (`training_dataset_id`) REFERENCES `training_dataset` (`id`) ON DELETE CASCADE ON UPDATE NO ACTION
+) ENGINE=ndbcluster DEFAULT CHARSET=latin1 COLLATE=latin1_general_cs;
+
+INSERT INTO `hopsworks`.`statistics_config` (`feature_group_id`, `descriptive`, `correlations`, `histograms`)
+SELECT `id` as `feature_group_id`,
+       `desc_stats_enabled` as `descriptive`,
+       `feat_corr_enabled` as `correlations`,
+       `feat_hist_enabled` as `histograms`
+    FROM `hopsworks`.`feature_group`;
+
+ALTER TABLE `hopsworks`.`feature_group`
+    DROP COLUMN `cluster_analysis_enabled`,
+    DROP COLUMN `num_clusters`,
+    DROP COLUMN `num_bins`,
+    DROP COLUMN `corr_method`,
+    DROP COLUMN `desc_stats_enabled`,
+    DROP COLUMN `feat_corr_enabled`,
+    DROP COLUMN `feat_hist_enabled`;
+
+INSERT INTO `hopsworks`.`statistics_config` (`training_dataset_id`, `descriptive`, `correlations`, `histograms`)
+SELECT `id` as `training_dataset_id`,
+       1 as `descriptive`,
+       0 as `correlations`,
+       0 as `histograms`
+    FROM `hopsworks`.`training_dataset`;
+
+CREATE TABLE `tmp_stats_columns` AS
+SELECT `configs`.`id` as `statistics_config_id`,
+       `columns`.`name` as `name`
+    FROM `hopsworks`.`statistic_columns` `columns`
+    INNER JOIN `hopsworks`.`statistics_config` `configs`
+    ON `columns`.`feature_group_id` = `configs`.`feature_group_id`;
+
+DROP TABLE `hopsworks`.`statistic_columns`;
+
+CREATE TABLE `statistic_columns` (
+  `id` int(11) NOT NULL AUTO_INCREMENT,
+  `statistics_config_id` int(11),
+  `name` varchar(500) COLLATE latin1_general_cs DEFAULT NULL,
+  PRIMARY KEY (`id`),
+  KEY `statistics_config_id` (`statistics_config_id`),
+  CONSTRAINT `statistics_config_fk` FOREIGN KEY (`statistics_config_id`) REFERENCES `statistics_config` (`id`) ON DELETE CASCADE ON UPDATE NO ACTION
+) ENGINE=ndbcluster DEFAULT CHARSET=latin1 COLLATE=latin1_general_cs;
+
+INSERT INTO `hopsworks`.`statistic_columns` (`statistics_config_id`, `name`)
+SELECT * FROM `hopsworks`.`tmp_stats_columns`;
+
+DROP TABLE `hopsworks`.`tmp_stats_columns`;
```

Comment:
This looks complicated. Here's what I'd do:
- add `statistics_config_id` to the `statistics_column` table.
- Update the rows so that `statistics_config_id` matches the correct value based on the `feature_group_id` already present
- Drop the `feature_group_id` column.

Reply:
simplified it as suggested.